### PR TITLE
Improve cloud-info-provider support

### PR DESCRIPTION
**Pull request by enolfc**
Add a debug option
Support provider using oidc instead of openid
Make sure things do not fail if not all secrets are avaiable

### CODE REVIEW

File: cloud-info/cloud-info-wrapper.sh [hunk 1]
```diff
@@ -1,9 +1,15 @@
 #!/bin/sh
 
+EXTRA_OPTS=""
+
+if test "xDEBUG" = "x1"; then
```
Comment (gwarf) [inline cloud-info/cloud-info-wrapper.sh]:
```suggestion
if test "x${DEBUG}" = "x1"; then
```

Comment (enolfc) [reply cloud-info/cloud-info-wrapper.sh]: well spotted, that explains why I was not getting the results I expected :)

Comment (gwarf) [reply cloud-info/cloud-info-wrapper.sh]: But honestly I always find this test quite ugly.
Usuall I prefer things relying on `-z` and `-n`.
So if you want to rely test the DEBUG value something like
```shell
if [ -n "$DEBUG" ] && [ "$DEBUG" = "1" ]; then
  EXTRA_OPTS="$EXTRA_OPTS --debug"
fi
```

File: cloud-info/cloud-info-wrapper.sh [hunk 2]
```diff
@@ -1,9 +1,15 @@
 #!/bin/sh
 
+EXTRA_OPTS=""
+
+if test "x${DEBUG}" = "x1"; then
```
Comment (gwarf) [inline cloud-info/cloud-info-wrapper.sh]: Honestly I always find this test quite ugly.
Usually I prefer things relying on `-z` and `-n`.
So if you want to rely test the DEBUG value something like
```shell
if [ -n "$DEBUG" ] && [ "$DEBUG" = "1" ]; then
  EXTRA_OPTS="$EXTRA_OPTS --debug"
fi
```

Comment (gwarf) [reply cloud-info/cloud-info-wrapper.sh]: and if we don't care about the exact value of `$DEBUG` but just want it to be set, `[ -n "$DEBUG" ]` may be enough.

Comment (enolfc) [reply cloud-info/cloud-info-wrapper.sh]: I guess we do not really care about the value, so test -n should be ok, I'm committing a change

**Pull request merged**